packages/@sanity/form-builder/src/inputs/ImageInput/Asset.js

```diff
@@ -0,0 +1,230 @@
+    const actions = []
+
+    if (dialogType != 'cantDelete') {
+      actions.push({name: 'delete', title: 'Delete', icon: TrashIcon, color: 'danger'})
+    }
+
+    actions.push({
+      name: 'close',
+      title: 'Close'
+    })
+
+    return actions
+  }
+
+  render() {
+    const {asset, onClick, onKeyPress} = this.props
+    const {isDeleting, showDialog, dialogType} = this.state
+    const size = 75
+    const width = get(asset, 'metadata.dimensions.width') || 100
+    const height = get(asset, 'metadata.dimensions.height') || 100
+
+    const menuItems = [
+      {
+        name: 'showRefs',
+        title: 'Show documents using this',
+        icon: LinkIcon
+      },
+      {
+        name: 'delete',
+        title: 'Delete',
+        color: 'danger',
+        icon: TrashIcon
+      }
+    ]
+
+    return (
+      <a
+        className={styles.item}
+        tabIndex={0}
+        data-id={asset._id}
+        onKeyPress={onKeyPress}
+        style={{
+          width: `${(width * size) / height}px`,
+          flexGrow: `${(width * size) / height}`
+        }}
+      >
+        <i className={styles.padder} style={{paddingBottom: `${(height / width) * 100}%`}} />
+        {/* We can not determine an alt text on image */}
+        <img
+          src={`${asset.url}?h=100`}
+          className={styles.image}
+          onClick={onClick}
+          data-id={asset._id}
+        />
+        {isDeleting && (
+          <div className={styles.spinnerContainer}>
+            <Spinner center />
+          </div>
+        )}
+        <div className={styles.menuContainer}>
+          <DropDownButton
+            placement="bottom-end"
+            showArrow={false}
+            items={menuItems}
+            renderItem={this.renderMenuItem}
+            onAction={this.handleMenuAction}
+          >
+            <MoreVertIcon />
+          </DropDownButton>
+          {showDialog && (
+            <Dialog
+              title={
+                dialogType === 'cantDelete'
+                  ? `Can't delete asset. Used in other documents`
+                  : 'Documents using this'
+              }
+              color={dialogType === 'cantDelete' ? 'danger' : undefined}
+              onClose={this.handleDialogClose}
+              onAction={this.handleDialogAction}
+              actions={this.getDialogActions(dialogType)}
+            >
+              <DialogContent size="medium">
+                <div className={styles.dialogContent}>
+                  <div>
+                    <img src={`${asset.url}?w=200`} />
+                  </div>
+                  <WithReferringDocuments id={asset._id}>
+                    {({isLoading, referringDocuments}) =>
+                      isLoading ? (
+                        <Spinner>Loading…</Spinner>
+                      ) : (
+                        <div>
+                          {referringDocuments.length === 0 ? (
+                            <div>No documents are using this asset</div>
```

Comment:
`No documents are referencing this asset`

Reply:
Same as above